ui-tests/cypress/integration/components/ux_fragments.spec.js

```diff
@@ -1,808 +1,665 @@
-        generateRandomFragment().then(fragment => {
-          openFragmentsPage()
-              .then(page =>
-                  page.getContent().openAddFragmentPage());
-
-          cy.get('@currentPage')
-            .then(page => {
-              page.getContent().getCodeInput().then(input => page.getContent().type(input, fragment.code));
-              page.getContent().getGuiCodeInput().then(input => page.getContent().type(input, fragment.guiCode));
-            })
-            .then(page => page.getContent().save())
-            .then(page => {
-              cy.validateToast(page);
-              cy.wrap(fragment).as('fragmentToBeDeleted');
-
-              page.getContent().getTableRows(fragment.code).should('be.visible')
-                  .children(htmlElements.td).then(cells =>
-                  cy.validateListTexts(cells, [fragment.code]));
-            });
-        });
-      });
-      it([Tag.SANITY, 'ENG-3522'], 'Editing a fragment', () => {
-        addRandomFragment().then(fragment => {
-          openFragmentsPage()
-              .then(page =>
-                  page.getContent().getKebabMenu(fragment.code).open().openEdit(fragment.code))
-              .then(page => page.getContent().getGuiCodeInput().then(input => page.getContent().type(input, fragment.guiCode)))
-              .then(page => page.getContent().save())
-              .then(page => {
-                cy.validateToast(page);
-                page.getContent().getTableRow(fragment.code).should('be.visible')
-                    .children(htmlElements.td).then(cells =>
-                    cy.validateListTexts(cells, [fragment.code, false]));
-              });
-        });
-      });
-      it([Tag.SANITY, 'ENG-3522'], 'Cloning', () => {
-        addRandomFragment().then(fragment => {
-          openFragmentsPage()
-              .then(page => {
-                page.getContent().getKebabMenu(fragment.code).open().openClone(fragment.code);
-                cy.wrap(fragment).as('fragmentToBeDeleted');
-              });
-
-          cy.get('@currentPage')
-            .then(page => {
-              page.getContent().getCodeInput().then(input => page.getContent().type(input, fragmentCloned.code));
-              page.getContent().getGuiCodeInput().then(input => page.getContent().type(input, fragmentCloned.guiCode));
-            })
-            .then(page => {
-              page.getContent().save();
-              cy.wrap(fragmentCloned).as('fragmentClonedToBeDeleted');
-            });
-          cy.get('@currentPage')
-            .then(page => {
-              cy.validateToast(page);
-              page.getContent().getTableRow(fragmentCloned.code).should('be.visible')
-                  .children(htmlElements.td).then(cells =>
-                  cy.validateListTexts(cells, [fragmentCloned.code, false]));
-            });
-          cy.get('@fragmentClonedToBeDeleted').then(fragmentToBeDeleted => {
```

Comment:
Perform the pre-requisite operation and the clean-up ones on _before_ and _after_, operation performed inside a test might not be executed